services/consensusalgo/leanhelixconsensus/network_communication.go

```diff
@@ -10,49 +10,42 @@ import (
 	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
 )
 
-func (s *service) HandleLeanHelixMessage(ctx context.Context, input *gossiptopics.LeanHelixInput) (*gossiptopics.EmptyOutput, error) {
-
-	message := leanhelix.CreateConsensusRawMessage(
-		leanhelix.MessageType(input.Message.MessageType),
-		input.Message.Content,
-		&BlockPairWrapper{
-			blockPair: input.Message.BlockPair,
-		},
-	)
+type networkCommunication struct {
+	gossip                  gossiptopics.LeanHelix
+	messageReceiversCounter int
+	messageReceivers        map[int]func(ctx context.Context, message leanhelix.ConsensusRawMessage)
+}
 
-	for _, messageReceiver := range s.messageReceivers {
-		messageReceiver(ctx, message)
+func NewNetworkCommunication(gossip gossiptopics.LeanHelix) *networkCommunication {
+	return &networkCommunication{
+		gossip:                  gossip,
+		messageReceivers:        make(map[int]func(ctx context.Context, message leanhelix.ConsensusRawMessage)),
+		messageReceiversCounter: 0,
 	}
-	return nil, nil
 }
 
-func (s *service) RequestOrderedCommittee(seed uint64) []lhprimitives.Ed25519PublicKey {
+func (comm *networkCommunication) RequestOrderedCommittee(seed uint64) []lhprimitives.Ed25519PublicKey {
 	panic("implement me")
 }
 
-func (s *service) IsMember(pk lhprimitives.Ed25519PublicKey) bool {
+func (comm *networkCommunication) IsMember(pk lhprimitives.Ed25519PublicKey) bool {
 	panic("implement me")
 }
 
 // Lib calls this method to register itself for incoming messages, and supplies the callback
-func (s *service) RegisterOnMessage(onReceivedMessage func(ctx context.Context, message leanhelix.ConsensusRawMessage)) int {
-	s.messageReceiversCounter++
-	s.messageReceivers[s.messageReceiversCounter] = onReceivedMessage
+func (comm *networkCommunication) RegisterOnMessage(onReceivedMessage func(ctx context.Context, message leanhelix.ConsensusRawMessage)) int {
```

Comment:
I'd prefer that you pass an interface with the function, to be consistent with our `FooHandler` convention

Reply:
Created "type messageHandler func(ctx context.Context, message leanhelix.ConsensusRawMessage)"
LMK if it's not what you meant.